File: ordenamiento_insercion/insercion.py
import random

# generar lista de los num aleatorios
arr = [random.randint(1, 100) for _ in range(10)]
print("Números random: ", arr)

# ordenamiento de insercion
for i in range(1, len(arr)):
    key = arr[i]
    j = i - 1
    while j >= o and key < arr[j]:
        arr[j + 1] = arr[j]
        j -= 1
    arr[j + 1] = key


print("Números random ordenados: ", arr)

File: ordenamiento_insercion/insercion.java
import java.util.Random;

public class InsertionSort {
    public static void main(String[] args) {
        Random rand = new Random();
        int[] arr = new int[10];

        // generar los num aleatorios
        for (int i = 0; i < arr.length; i++) {
            arr[i] = rand.nextInt(100) + 1;
        }

        System.out.print("Números random: ");
        for (int num : arr) System.out.print(num + " ");
        System.out.println();

        // ordenamiento de insercion
        for (int i = 1; i < arr.length; i++) {
            int key = arr[i];
            int j = i - 1;
            while (j >= 0 && arr[j] > key) {
                arr[j + 1] = arr[j];
                j--;
            }
            arr[j + 1] = key;
        }

        System.out.print("Números random ordenados: ");
        for (int num : arr) System.out.print(num + " ");
    }
}